native/src/main/java/io/ballerina/stdlib/websocket/WebSocketService.java

```diff
@@ -55,6 +67,37 @@ private void populateResourcesMap(BObject service) {
         }
     }
 
+    private Map<String, MethodType> getDispatchingFunctionMap(Object dispatchingService) {
+        Map<String, MethodType> dispatchingFunctions = new ConcurrentHashMap<>();
+        Set<String> seenRemoteFunctionNames = new HashSet<>();
+        MethodType[] remoteFunctions = ((ServiceType) (((BValue) dispatchingService).getType())).getMethods();
+        for (MethodType remoteFunc : remoteFunctions) {
```

Comment:
Is the variable nam `remoteFunctions` accurate here? This list contains all the methods of the service type, right? Maybe we can name this as `methods`?

Can't we just pass the `ServiceType` here, instead of passing it as an Object, since we don't use `dispatchingService` anywhere else? It will reduce the complexity of this code:

I think we should also omit checking the methods other than remote methods. You can check this by using the `qualifiers`.

```suggestion
    private Map<String, MethodType> getDispatchingFunctionMap(ServiceType dispatchingService) {
        Map<String, MethodType> dispatchingFunctions = new ConcurrentHashMap<>();
        Set<String> seenRemoteFunctionNames = new HashSet<>();
        for (MethodType remoteFunc : dispatchingService.getMethods()) {
```

Reply:
Updated via https://github.com/ballerina-platform/module-ballerina-websocket/pull/1553/commits/6d4fc9045dc30cb81e87f10adbba594be382e3f7 and https://github.com/ballerina-platform/module-ballerina-websocket/pull/1553/commits/df9da2d43f95ab57ec0215e0841f3d7f55be9e38